src/Components/HistoricalDataChart/Util.js

```diff
@@ -13,7 +13,7 @@ export const formatHistoricalData = (responseData, dateRange) =>{
 
     let mapResponseData = new Map();
 
-    responseData.forEach((responseValue) => {
+    responseData.reverse().forEach((responseValue) => {
         mapResponseData.set(new Date(responseValue.report_date).toDateString(), responseValue);
     });
```

Comment:
```suggestion
    responseData.forEach((responseValue) => {
      const report_date_val = new Date(responseValue.report_date).toDateString();
      if (mapResponseData.has(report_date_val)) {
          return;
      }
      mapResponseData.set(report_date_val, responseValue);
    });
```

Reply:
> responseData.reverse().forEach((responseValue)

@PreetiW `responseData` is sorted by report_date (not by cpu/memory values). We'll face the issue even after adding the last data (recent data) to the `mapResponseData` as we can have random values for cpu/memory.

Reply:
> if (mapResponseData.has(report_date_val)) {
          return;
      }

@kgaikwad Are we just considering the lastest data?

Reply:
@PreetiW - ignore my comment, my bad! I thought it might keep duplicate entries and next loop might give wrong result.

Reply:
> @kgaikwad Are we just considering the lastest data?

Yes @r14chandra  for now. If response includes multiple points info on a day, graph should show single latest point information. Does it make sense?

Reply:
Then the graph can be seem as `idle` (or any other) if we have latest data which fulfills the condition for Idling state.

Reply:
> responseData.forEach((responseValue) => {
      const report_date_val = new Date(responseValue.report_date).toDateString();
      if (mapResponseData.has(report_date_val)) {
          return;
      }
      mapResponseData.set(report_date_val, responseValue);
    });

If we're considering latest data then we don't need to iterate the whole response data - we can optimize it. Just pick the first.

Reply:
@r14chandra even if response data map will have random data it wont affect the data we are generating for graph as we are going to call .get on `responseData` Map. 

reverse will map over the data in reverse order so the old record value will get override with the latest record value.
I hope that answers your question.

Reply:
Yeah then I didn't know that we want to use the latest data only 👍🏻 .